src/templates/build-debug/build-debug-ios.ejf

```diff
@@ -0,0 +1,68 @@
+<%
+const isMonorepo = props.pathRelativeToRoot !== '.'
+
+const install = {
+  yarn: "yarn install --immutable",
+  npm: "npm ci"
+}
+
+const build = {
+  yarn: `yarn ${(isMonorepo) ? `--cwd ${props.pathRelativeToRoot} ` : ""}build:debug:ios`,
+  npm: `npm run build:debug:ios ${(isMonorepo) ? `--prefix ${props.pathRelativeToRoot}` : ""}`
+}
+%>
+
+name: Build iOS Debug App
+
+on:
+  workflow_call:
+    outputs:
+      build-cache-key:
+        description: 'The primary key to used for caching the build'
+        value: ios-debug-build-${{ jobs.lookup-cached-build.outputs.fingerprint }}
+
+jobs:
+  lookup-cached-build:
+    name: Lookup cached debug build
+    uses: ./.github/workflows/<%= props.lookupDebugBuildWorkflowFileName %>
+    with:
+      platform: ios
+
+  build-debug-ios:
+    name: Build iOS Debug App
+    needs: lookup-cached-build
+    if: needs.lookup-cached-build.outputs.build-exists != 'true'
+    runs-on: macos-latest
+    steps:
+      - name: 🏗 Checkout repository
+        uses: actions/checkout@v4
+
+      - name: 🌿 Setup Node 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: '<%= props.packageManager %>'
+
+      - name: 🔨 Use latest stable Xcode
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: latest-stable
+
+      - name: 📦 Install dependencies
+        run: <%= install[props.packageManager] %>
+
+      - name: 🛠️ Build
+        run: <%= build[props.packageManager] %>
+
+      - name: 📁 Prepare cache folder
+        run: |
+          mkdir ios-debug-build
+          mv <%= props.pathRelativeToRoot %>/ios/build/Build/Products/Debug-iphonesimulator/<%= props.iOSAppName %>.app ios-debug-build/ios-debug.app
+
+      - name: 📡 Store built app in cache
+        id: ios-debug-save
+        uses: actions/cache/save@v4
+        with:
+          path: ios-debug-build/
+          key: ios-debug-build-${{ needs.lookup-cached-build.outputs.fingerprint }}
```

Comment:
I wonder if running build for ios won't change the finger print. For example pod install may generate Podfile.lock

Reply:
you mean after `expo prebuild`? in this case (the workflow i managed), directories `ios/` and `android` are ignored when calculating fingerprint:

https://github.com/expo/expo/blob/ee198d023415c5d3e0f9ae9676f567badb48821d/packages/expo-updates/utils/src/createFingerprintAsync.ts#L16-L24